platform/darwin/src/MGLBaseStyleLayer.h

```diff
@@ -4,17 +4,24 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/**
+ The base style layer class from which all other layer types inherit. 
+ Style layers allow runtime customization of all map styling properties. 
+ */
 @interface MGLBaseStyleLayer : NSObject
```

Comment:
I think it'd be a good idea to, like we do for `MGLMultiPoint`, mention that this is an abstract class and that subclasses should be used in practice. Similarly, we should document that using the properties of this class is intended.